src/red/java/com/lambdaherding/edi/red/ch02/Calculator.java

```diff
@@ -0,0 +1,174 @@
+package com.lambdaherding.edi.red.ch02;
+
+import java.awt.BorderLayout;
+import java.awt.GridLayout;
+import java.util.Stack;
+import java.util.function.BinaryOperator;
+import java.util.function.UnaryOperator;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+
+import com.lambdaherding.edi.ch02.Calculate;
+
+public class Calculator extends JFrame {
+
+	private static final long serialVersionUID = 3809823621528950529L;
+
+	private JTextField display;
+	private StringBuilder number = new StringBuilder();
+	private Stack<Double> data = new Stack<>();
+
+	public Calculator() {
+		super( "My Calculator" );
+		setDefaultCloseOperation( JFrame.EXIT_ON_CLOSE );
+
+		setLayout( new BorderLayout() );
+
+		display = new JTextField();
+		display.setEditable( false );
+		add( display, BorderLayout.NORTH );
+
+		JPanel buttonPanel = makeButtons();
+		add( buttonPanel, BorderLayout.CENTER );
+
+		pack();
+	}
+
+	private JPanel makeButtons() {
+		JPanel pan = new JPanel();
+		pan.setLayout( new GridLayout( 4, 5 ) );
+
+		pan.add( inputButton( "7" ) );
+		pan.add( inputButton( "8" ) );
+		pan.add( inputButton( "9" ) );
+		pan.add( binaryActionButton( "/", (a, b) -> 1 / a * b ) );
```

Comment:
Hmm, well, that would seem to be the latter value divided by the former value, which sounds odd to me. But ehhhh.